search-app/src/cmr/search/services/community_usage_metrics/metrics_service.clj

```diff
@@ -130,28 +140,33 @@
     (errors/throw-service-error :invalid-data "You posted empty content")))
 
 (defn get-community-usage-metrics
-  "Retrieves the set of community usage metrics from metadata-db."
+  "Retrieves the current community usage metrics from metadata-db."
   [context]
   (:community-usage-metrics (json/decode (:metadata (humanizer-service/fetch-humanizer-concept context)) true)))
 
+(defn- community-usage-metrics-list->set
+  "Convert list of {:short-name :access-count} maps into a set of short-names."
+  [metrics-list]
+  (set (for [metric-map metrics-list]
+         (:short-name metric-map))))
+
 (defn- community-usage-csv->community-usage-metrics
-  "Validate the community usage csv and convert to a list of community usage metrics to save"
-  [context community-usage-csv comprehensive]
-  (let [{:keys [csv-lines product-col version-col hosts-col]} (validate-and-read-csv community-usage-csv)
-        current-metrics (when (= "false" comprehensive)
-                          (get-community-usage-metrics context))]
-    (map #(csv-entry->community-usage-metric context % product-col version-col hosts-col comprehensive current-metrics)
+  "Validate the community usage csv and convert to a list of community usage metrics."
+  [context community-usage-csv current-metrics]
+  (let [{:keys [csv-lines product-col hosts-col]} (validate-and-read-csv community-usage-csv)
+        short-name-cache (new java.util.HashMap)]
+    (map #(csv-entry->community-usage-metric context % product-col hosts-col current-metrics short-name-cache)
          csv-lines)))
 
 (defn- aggregate-usage-metrics
-  "Combine access counts for entries with the same short-name and version number."
+  "Combine access counts for entries with the same short-name."
   [metrics]
-  ;; name-version-groups is map of [short-name, version] [entries that match short-name/version]
-  (let [name-version-groups (group-by (juxt :short-name :version) metrics)] ; Group by short-name and version
+  ;; name-groups is map of short-names [entries that match short-name]
+  (let [name-groups (group-by (juxt :short-name) metrics)] ; Group by short-name and version
```

Comment:
Are we supposed to drop the `version`? If so, we don't need the `juxt` call, just `(group-by :short-name metrics)` will do.